AutoCollection/AzureFunctionsHook.ts

```diff
@@ -109,17 +109,21 @@ export class AzureFunctionsHook {
 
     private _createIncomingRequestTelemetry(request: HttpRequest, response: HttpResponse, startTime: number, parentId: string) {
         let statusCode: string | number = 200; //Default
-        for (const value of [response.statusCode, response.status]) {
-            if (typeof value === "number" && Number.isInteger(value)) {
-                statusCode = value;
-                break;
-            } else if (typeof value === "string") {
-                const parsedVal = parseInt(value);
-                if (!isNaN(parsedVal)) {
-                    statusCode = parsedVal;
+        if (response) {
+            for (const value of [response.statusCode, response.status]) {
+                if (typeof value === "number" && Number.isInteger(value)) {
+                    statusCode = value;
                     break;
+                } else if (typeof value === "string") {
+                    const parsedVal = parseInt(value);
+                    if (!isNaN(parsedVal)) {
+                        statusCode = parsedVal;
+                        break;
+                    }
                 }
             }
+        } else {
+            statusCode = 500;
```

Comment:
I think it would be better to have an empty statusCode here, we don't want to assume the code if response is not even there

Reply:
Makes sense. I'll leave it undefined and update the `statusCode` check.